### PR TITLE
Bank Login

### PR DESCRIPTION
**Pull request by martinmilev**
[presentation](https://martinmilev.github.io/htmlPages/servlets/bankLog/bank2.html)
Ready for review.

### CODE REVIEW

File: src/main/java/com/clouway/http/servlets/IndexPageServlet.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  public IndexPageServlet(AccountRepository repository, Template template) {
+    this.repository = repository;
+    this.template = template;
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String param = req.getParameter("link");
+    if ("login".equals(param)) {
+      resp.sendRedirect("/login");
```
Comment (freekman) [inline src/main/java/com/clouway/http/servlets/IndexPageServlet.java]: is this needed

Comment (freekman) [inline src/main/java/com/clouway/http/servlets/IndexPageServlet.java]: why does the index page need to redirect

File: src/main/java/com/clouway/http/servlets/LoginPageServlet.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    getInnerHtml(resp, HttpServletResponse.SC_OK, template.evaluate());
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String name = req.getParameter("name");
+
+    if (repository.getByName(name).isPresent()) {
```
Comment (freekman) [inline src/main/java/com/clouway/http/servlets/LoginPageServlet.java]: so as i can see you are logging only with the name , you dont  need the pwd and then what happens noting

File: src/main/test/com/clouway/persistent/adapter/jdbc/PersistentAccountRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  @Before
+  public void setUp() throws Exception {
+    ConnectionProvider provider = new ConnectionProvider("jdbc:mysql://localhost/myBank", "root", "123123");
+    DataStore dataStore = new DataStore(provider);
+    repo = new PersistentAccountRepository(dataStore);
+    dataStore.update("truncate accounts");
+  }
+
+//  @Test
```
Comment (freekman) [inline src/main/test/com/clouway/persistent/adapter/jdbc/PersistentAccountRepositoryTest.java]: ?

File: src/main/java/com/clouway/persistent/datastore/RowFetcher.java [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
 
 import java.sql.ResultSet;
+import java.sql.SQLException;
 
 /**
  * @author Martin Milev <martinmariusmilev@gmail.com>
  */
 public interface RowFetcher<T> {
-  T fetchRow(ResultSet resultSet);
+  T fetchRow(ResultSet resultSet) throws SQLException;
```
Comment (freekman) [inline src/main/java/com/clouway/persistent/datastore/RowFetcher.java]: .

File: src/main/resources/index.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
Comment (freekman) [inline src/main/resources/index.html]: why is this html added to the PR

File: src/main/resources/login.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+</head>
+<body>
+<h1 class="text-center">myBank</h1>
+<hr>
+
+<div class="container-fluid col-md-2 col-md-offset-5">
+    <nav class="navbar">
+        <ul class="nav nav-tabs">
+            <li role="presentation"><a href="/?link=register">Register</a></li>
```
Comment (freekman) [inline src/main/resources/login.html]: not needed ```/?link=register"```

File: src/main/resources/login.html [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            <li role="presentation" class="active"><a href="#">Login</a></li>
+        </ul>
+    </nav>
+    <form class="form-horizontal" action="/" method='post'>
```
Comment (freekman) [inline src/main/resources/login.html]: why action is ```/```

File: src/main/test/com/clouway/http/servlets/IndexPageServletTest.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
+package com.clouway.http.servlets;
```
Comment (freekman) [inline src/main/test/com/clouway/http/servlets/IndexPageServletTest.java]: unneded for this PR

File: src/main/test/com/clouway/persistent/adapter/jdbc/PersistentAccountRepositoryTest.java (hunk 3 above)
Comment (martinmilev) [inline src/main/test/com/clouway/persistent/adapter/jdbc/PersistentAccountRepositoryTest.java]: Don't know why it is in that PR.

File: src/main/resources/index.html (hunk 5 above)
Comment (martinmilev) [inline src/main/resources/index.html]: It contains changes.

File: src/main/java/com/clouway/http/servlets/IndexPageServlet.java (hunk 1 above)
Comment (martinmilev) [inline src/main/java/com/clouway/http/servlets/IndexPageServlet.java]: It is a link. It is needed if you want to go to the login page.

File: src/main/resources/login.html (hunk 6 above)
Comment (martinmilev) [inline src/main/resources/login.html]: User may want to switch between login and register.

File: src/main/resources/login.html (hunk 7 above)
Comment (martinmilev) [inline src/main/resources/login.html]: Breaking my code. Any action other than "#" does it.

File: src/main/test/com/clouway/http/servlets/IndexPageServletTest.java (hunk 8 above)
Comment (martinmilev) [inline src/main/test/com/clouway/http/servlets/IndexPageServletTest.java]: It contains changes.

File: src/main/resources/login.html [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+</head>
+<body>
+<h1 class="text-center">myBank</h1>
+<hr>
+
+<div class="container-fluid col-md-2 col-md-offset-5">
+    <nav class="navbar">
+        <ul class="nav nav-tabs">
+            <li role="presentation"><a href="/register">Register</a></li>
+            <li role="presentation" class="active"><a href="/register">Login</a></li>
```
Comment (freekman) [inline src/main/resources/login.html]: .

**Comment by freekman** [comment]
LGTM after you fix last comment

**Pull request merged**